logstash-core/src/main/java/org/logstash/config/ir/compiler/EventCondition.java

```diff
@@ -67,7 +67,7 @@
         /**
          * Cache of all compiled {@link EventCondition}.
          */
-        private static final Map<String, EventCondition> CACHE = new HashMap<>(10);
+        private static final Map<String, EventCondition> CACHE = new ConcurrentHashMap<>(10);
```

Comment:
Thinking about this, I'd prefer we use an LRU here. With config reloading we have a leak here.

It's not strictly in scope for this PR. I've opened https://github.com/elastic/logstash/issues/9273 to track it.